src/_pytest/recwarn.py

```diff
@@ -286,11 +289,8 @@ def __exit__(
                                 break
                     else:
                         fail(
-                            "DID NOT WARN. No warnings of type {} matching"
-                            " ('{}') were emitted. The list of emitted warnings"
-                            " is: {}.".format(
-                                self.expected_warning,
-                                self.match_expr,
-                                [each.message for each in self],
-                            )
+                            f"""\
+DID NOT WARN. No warnings of type {self.expected_warning} matching the regex were emitted.
+The regex is: {self.match_expr}
+The list of emitted warnings is: {found_str()}"""
```

Comment:
```suggestion
 Regex: {self.match_expr}
 Emitted warnings: {found_str()}"""
```

for a bit less output and more consistency with `match` above